omod/src/main/java/org/openmrs/module/smartonfhir/web/filter/SmartSessionLogoutFilter.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.smartonfhir.web.filter;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.keycloak.adapters.AdapterDeploymentContext;
+import org.keycloak.adapters.KeycloakDeploymentBuilder;
+import org.keycloak.adapters.spi.HttpFacade;
+import org.keycloak.adapters.springsecurity.facade.SimpleHttpFacade;
+import org.openmrs.module.smartonfhir.util.KeycloakConfigHolder;
+
+@Slf4j
+public class SmartSessionLogoutFilter implements Filter {
+	
+	private volatile String logoutUrl = null;
+	
+	private volatile AdapterDeploymentContext adapterDeploymentContext = null;
```

Comment:
This probably doesn't need to be `volatile`. This is because the Servlet spec guarantees that `init()` (where we setup `adapterDeploymentContext` will be called only once from one thread. On the other hand, `logoutUrl`, which we appear to need a request and response to initialize could be called from multiple threads at once.